plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/Processor.java

```diff
@@ -51,7 +67,10 @@
          * Implementations are responsible for removing the used keys, so that after creating a pipeline ingest can
          * verify if all configurations settings have been used.
          */
-        P create(Map<String, Object> config) throws Exception;
+        P create(String processorId, Map<String, Object> config) throws Exception;
```

Comment:
I think we should have one `create(...)` method here? If there is no processor id then just `null` is provided.